graph/telemetry/istio/appender/appender.go

```diff
@@ -316,3 +340,40 @@ func getAppWorkloads(namespace, app, version string, gi *graph.AppenderGlobalInf
 	}
 	return result
 }
+
+func getApp(namespace, appName string, gi *graph.AppenderGlobalInfo) (*models.AppListItem, bool) {
+	if appName == "" || appName == graph.Unknown {
+		return nil, false
+	}
+
+	var allAppsMap map[string]appsMap
+	if existingAllAppsMap, ok := gi.Vendor[appsMapKey]; ok {
+		allAppsMap = existingAllAppsMap.(map[string]appsMap)
+	} else {
+		allAppsMap = make(map[string]appsMap)
+		gi.Vendor[appsMapKey] = allAppsMap
+	}
+
+	var namespaceApps appsMap
+	if existingNamespaceApps, ok := allAppsMap[namespace]; ok {
+		if app, ok := existingNamespaceApps[appName]; ok {
+			return app, true
+		} else {
+			namespaceApps = existingNamespaceApps
+		}
+	} else {
+		namespaceApps = appsMap{}
+		allAppsMap[namespace] = namespaceApps
+	}
+
+	if appList, err := gi.Business.App.GetAppList(context.TODO(), business.AppCriteria{Namespace: namespace, IncludeIstioResources: false}); err == nil {
```

Comment:
This is good.
@aljesusg is also adding a "criteria" here https://github.com/kiali/kiali/pull/4763/files#diff-80818359e8256dc52cb343d2aa947665c308d27ffb25b1930707e115ded5061fR27 that should have also an on/off flag to not include it in the graph.

But with this object then adjusting what that method may return would be easier.

Reply:
If health is not true then api not return the health info

Reply:
The labeler appender does not need health.